### ISSUE TITLE
Add tracking DB to current Pushmi-Pullyu structure

### ISSUE THREAD

**Issue by weiweishi** [ualbertalib/pushmi_pullyu#91]
This is to prepare for the phase 2 of the preservation tasks that are based on file system (non-Fedora 4/ERA triggered) with PMPY. 

This is needed for projects such as the Qureshi and Imanishi Data rescue project, Peel/Digitization materials ingest, and potential CWRC. 

Requirements:
- [ ] Add a DB for PushMi-PullYu to track information includes:
  - Noid
  - the path to the object or the equivalent of the "ID" of the item within the project as identified by the script
  - Timestamp of the ingest
  - Project name
  - File size of the data in the AIP
  - File count of the different formats of the data in the AIP (as a text blob eg: .WAV: 3 files, .XML: 5 files etc)
  - Checksum

A sample of Qureshi object as a test case.

**Comment by murny** [comment]
Alll this information could be tracked in the log file just like how PMPY currently does it.

So my questions are, Is this to replace the log file for PMPY? Will this DB information eventually be exposed to a web UI interface? Or why this requirement?

**Comment by weiweishi** [comment]
Tracking the information especially the noid and item identifier within the project, and being able to search and aggregate the information based on project is very critical for non-fedora objects, as we don't have any other system such as Fedora to track that information. This will potentially be our SSOT for all the digital content we preserve.  

My proposal is to replace the log file with a DB, for its searchability, and build an admin UI to expose that information down the road. And maybe this could be the base of our DAMS? thoughts?

**Comment by murny** [comment]
This makes sense and we can do this (this was discussed many times when building PushmiPullyu but was always deemed to be a nice to have and we decided to use logs for the MVP). 

As mentioned after standup however, I think we have to be careful where this Admin UI web interface lives though with the harsh security requirements around PushmiPullyu. Probably easiest is have PushmiPullyu push data into a DB, but have a different app (probably rails app since this is most familiar on our team) consume the DB outside the firewall and present this information to the web with an Admin UI. I highly doubt this UI could live in PushmiPullyu because of the strict firewalls. Even if it could, I think its really silly and not very useful to have a website that can only be accessed from a particular work station during work hours.

So some food for thought and definately need more thinking regarding this.

**Comment by weiweishi** [comment]
Yes, thanks! I should be more clear about the admin UI. My thinking is along the same line as yours, - a web interface built on top of this tracking/logging database, probably with RoR, not in scope with PMPY, but a separate project, and it could be served as the base for a DAMS and potentially as part of the Fedora exit strategy?

Lots to think indeed.
